cli/azd/internal/appdetect/spring_boot.go

```diff
@@ -241,15 +241,48 @@ func detectStorageAccountAccordingToSpringCloudStreamBinderMavenDependencyAndPro
 	}
 }
 
-func detectSpringCloudAzure(azdProject *Project, springBootProject *SpringBootProject) {
+func detectMetadata(azdProject *Project, springBootProject *SpringBootProject) {
+	detectDependencySpringCloudAzureStarter(azdProject, springBootProject)
+	detectDependencySpringCloudAzureStarterJdbcPostgresql(azdProject, springBootProject)
+	detectDependencySpringCloudAzureStarterJdbcMysql(azdProject, springBootProject)
+	detectPropertySpringDatasourcePassword(azdProject, springBootProject)
+}
+
+func detectDependencySpringCloudAzureStarter(azdProject *Project, springBootProject *SpringBootProject) {
 	var targetGroupId = "com.azure.spring"
 	var targetArtifactId = "spring-cloud-azure-starter"
 	if hasDependency(springBootProject, targetGroupId, targetArtifactId) {
-		azdProject.MetaData.ContainsDependencySpringCloudAzureStarter = true
+		azdProject.Metadata.ContainsDependencySpringCloudAzureStarter = true
 		logMetadataUpdated("ContainsDependencySpringCloudAzureStarter = true")
 	}
 }
 
+func detectDependencySpringCloudAzureStarterJdbcPostgresql(azdProject *Project, springBootProject *SpringBootProject) {
+	var targetGroupId = "com.azure.spring"
+	var targetArtifactId = "spring-cloud-azure-starter-jdbc-postgresql"
+	if hasDependency(springBootProject, targetGroupId, targetArtifactId) {
+		azdProject.Metadata.ContainsDependencySpringCloudAzureStarterJdbcPostgresql = true
+		logMetadataUpdated("ContainsDependencySpringCloudAzureStarterJdbcPostgresql = true")
+	}
+}
+
+func detectDependencySpringCloudAzureStarterJdbcMysql(azdProject *Project, springBootProject *SpringBootProject) {
+	var targetGroupId = "com.azure.spring"
+	var targetArtifactId = "spring-cloud-azure-starter-jdbc-mysql"
+	if hasDependency(springBootProject, targetGroupId, targetArtifactId) {
+		azdProject.Metadata.ContainsDependencySpringCloudAzureStarterJdbcMysql = true
+		logMetadataUpdated("ContainsDependencySpringCloudAzureStarterJdbcMysql = true")
+	}
+}
+
+func detectPropertySpringDatasourcePassword(azdProject *Project, springBootProject *SpringBootProject) {
+	var targetProperty = "spring.datasource.password"
```

Comment:
Hi, @saragluna , @moarychan

Whether `spring.datasource.password` should be deleted in customer's application.yml when using passwordless connection?

1. Not need to delete. I tested deploy by mi, keeping this property will not cause problem.
2. Not need to delete. I checked the source code: is empty will continue the logic of passwordless.
![image](https://github.com/user-attachments/assets/c5209b58-92b6-4b00-b5fa-a25308e2e53d)

3. It's necessary to delete. According to this doc: https://learn.microsoft.com/en-us/azure/developer/java/spring-framework/migrate-postgresql-to-passwordless-connection?tabs=sign-in-azure-cli%2Cspring%2Capp-service%2Cassign-role-service-connector
> ![image](https://github.com/user-attachments/assets/633d0731-a513-42bf-87d9-280ab0715a0b)

4. I remember it's necessary to delete it because empty string is also a valid password for DB.

Reply:
We provide a property isPasswordlessEnabled, it should be met first. If use uses an empty password to connection db, then the isPasswordlessEnabled should be set to false, is this OK?